apps/arena/lib/arena/game/effect.ex

```diff
@@ -26,16 +26,30 @@ defmodule Arena.Game.Effect do
   def put_effect_to_entity(game_state, entity, owner_id, start_action_removal_in_ms, effect) do
     last_id = game_state.last_id + 1
 
-    entity_contain_effect? =
-      Enum.any?(entity.aditional_info.effects, fn entity_effect ->
+    same_applied_effect =
+      Enum.find(entity.aditional_info.effects, fn entity_effect ->
         entity_effect.owner_id == owner_id and entity_effect.name == effect.name
       end)
 
+    now = System.monotonic_time(:millisecond)
+
     updated_entity =
-      if entity_contain_effect? and effect.one_time_application do
-        entity
+      if same_applied_effect && effect.one_time_application do
+        updated_effects =
+          Enum.map(entity.aditional_info.effects, fn effect ->
+            if effect.id == same_applied_effect.id do
+              refresh_effect_expires_at(same_applied_effect, now)
+            else
+              effect
+            end
+          end)
+
+        put_in(
+          entity,
+          [:aditional_info, :effects],
+          updated_effects
+        )
```

Comment:
If you use `and` this whole block could be the private function.

Reply:
yeah i think both if path can be replaced with functions, in fact i just discovered a bug, we were always increasing the game_state.last_id even when we were refreshing the expire_time. I just created two private functions for each path